src/AppBundle/Service/OrderManager.php

```diff
@@ -186,20 +190,19 @@ public function capturePayment(OrderInterface $order)
         $stateMachine = $this->stateMachineFactory->get($stripePayment, PaymentTransitions::GRAPH);
 
         $stripeAccount = $order->getRestaurant()->getStripeAccount();
+        $restaurantPaysStripeFee = $order->getRestaurant()->getContract()->isRestaurantPaysStripeFee();
+        $stripeOptions = array();
 
-        if (!is_null($stripeAccount)) {
-            $restaurantStripeId = $stripeAccount->getStripeUserId();
-        } else {
-            $restaurantStripeId = null;
+        if (!is_null($stripeAccount) && $restaurantPaysStripeFee) {
+            $stripeOptions['stripe_account'] = $stripeAccount->getStripeUserId();
         }
 
         try {
 
             $charge = Stripe\Charge::retrieve(
                 $stripePayment->getCharge(),
-                array(
-                    'stripe_account' => $restaurantStripeId
-            ));
+                $stripeOptions
```

Comment:
Can you remove this extra-param (and code related to it), as we don't need it to retrieve the charge ?

Reply:
I removed the param and tried it out but the charge is not retrieved correctly because it is not associated with the coop's Stripe ID. We have to pass the restaurant's Stripe ID when using Direct Charges.

Reply:
I gave it more thought, and it actually makes perfect sense. When the restaurant pays the fee, the created Stripe charge is not linked to our account at all (actually you don't see anything from the platform's stripe dashboard before capture and seeing the application fee then). you need the extra-parameter to "connect as" the restaurant owner.

Reply:
Yes, exactly.